src/primitives/line/dotted_bresenham.rs

```diff
@@ -0,0 +1,246 @@
+///
+/// [`DottedBresenhamParameters`] describes a scalar major step and a
+/// scalar minor step. It is used to draw a dotted line with the
+/// desired number of dots by selecting the appropriate points in a
+/// bresenham line.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(feature = "defmt", derive(::defmt::Format))]
+pub struct DottedBresenhamParameters {
+    /// Error threshold.
+    ///
+    /// If the accumulated error exceeds the threshold a minor move is made.
+    pub error_threshold: i32,
+
+    /// Change in error for major and minor steps.
+    pub error_step: MajorMinor<i32>,
+
+    /// Change in index for major and minor steps.
+    pub index_step: MajorMinor<usize>,
+}
+
+impl DottedBresenhamParameters {
+    /// Create a new bresenham parameters object.
+    pub fn new(line: &Line, nb_dots_desired: i32) -> Self {
+        let delta = line.delta().abs();
+        let line_max_index = delta.x.max(delta.y);
+
+        // Enforce the use of at most `line_max_index + 1` dots.
+        let mut nb_dots = min(nb_dots_desired, line_max_index + 1);
+        // Then enforce the use of at least 2 dots to prevent division by 0 when the line is reduced to a point.
+        nb_dots = max(2, nb_dots);
+
+        let integer_quotient = line_max_index / (nb_dots - 1);
+        let remainder = line_max_index - integer_quotient * (nb_dots - 1);
+
+        let index_step = MajorMinor::new(integer_quotient as usize, 1);
+        let error_threshold = nb_dots - 1;
+
+        Self {
+            error_threshold,
+            error_step: MajorMinor::new(remainder * 2, error_threshold * 2),
+            index_step,
+        }
+    }
+
+    /// Increases the error by a major step.
+    ///
+    /// If the error threshold is reached the error is reduced by a minor step and
+    /// `true` is returned.
+    pub fn increase_error(&self, error: &mut i32) -> bool {
+        *error += self.error_step.major;
+        if *error >= self.error_threshold {
+            *error -= self.error_step.minor;
+
+            true
+        } else {
+            false
+        }
+    }
+}
+
+/// Bresenham algorithm for dotted lines.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(feature = "defmt", derive(::defmt::Format))]
+pub struct DottedBresenham {
+    /// Current index increment.
+    ///
+    /// It is used to retrieve the next dot from a [`Points`] iterator.
+    pub index_nth: Option<usize>,
+
+    /// Error accumulator.
+    error: i32,
+}
+
+impl DottedBresenham {
+    /// Create a new bresenham object.
+    pub const fn new(initial_index: usize) -> Self {
+        Self::with_initial_error(initial_index, 0)
+    }
+
+    /// Create a new extended bresenham object with the initial error.
+    pub const fn with_initial_error(initial_index: usize, initial_error: i32) -> Self {
+        Self {
+            index_nth: Some(initial_index),
+            error: initial_error,
+        }
+    }
```

Comment:
There's no need to keep the `with_initial_error` constructor if it isn't used:

```suggestion
    /// Create a new bresenham object.
    pub const fn new(initial_index: usize) -> Self {
        Self {
            index_nth: Some(initial_index),
            error: 0,
        }
    }
```